ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -101,6 +101,13 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 	rayDashboardClient := utils.GetRayDashboardClientFunc()
 	rayDashboardClient.InitClient(clientURL)
 
+	// Check the current status of ray cluster before submitting.
+	if rayClusterInstance.Status.State != rayv1alpha1.Ready {
+		r.Log.Info("waiting for the cluster to be ready", "rayCluster", rayClusterInstance.Name)
+		err = r.updateState(ctx, rayJobInstance, rayv1alpha1.JobDeploymentStatusInitializing, nil)
+		return ctrl.Result{}, err
```

Comment:
minor: We can requeue it after x seconds since cluster ready takes a while. without requeueAfter we will see lots of duplicated logs. 

This is minor, I can update it later in another PR